src/hotspot/share/cds/filemap.cpp

```diff
@@ -2090,22 +2021,18 @@ MemRegion FileMapInfo::get_heap_regions_requested_range() {
   address start = (address) max_uintx;
   address end   = nullptr;
 
-  for (int i = MetaspaceShared::first_closed_heap_region;
-           i <= MetaspaceShared::last_valid_region;
-           i++) {
-    FileMapRegion* r = region_at(i);
-    size_t size = r->used();
-    if (size > 0) {
-      address s = heap_region_requested_address(r);
-      address e = s + size;
-      log_info(cds)("Heap region %s = " INTPTR_FORMAT " - " INTPTR_FORMAT " = "  SIZE_FORMAT_W(8) " bytes",
-                    region_name(i), p2i(s), p2i(e), size);
-      if (start > s) {
-        start = s;
-      }
-      if (end < e) {
-        end = e;
-      }
+  FileMapRegion* r = region_at(MetaspaceShared::hp);
+  size_t size = r->used();
+  if (size > 0) {
+    address s = heap_region_requested_address(r);
+    address e = s + size;
+    log_info(cds)("Heap region = " INTPTR_FORMAT " - " INTPTR_FORMAT " = "  SIZE_FORMAT_W(8) " bytes",
+                  p2i(s), p2i(e), size);
+    if (start > s) {
+      start = s;
+    }
+    if (end < e) {
+      end = e;
```

Comment:
Like mentioned before, maybe we have max() and min() methods to use here.

Reply:
I simplified the code -- there's only one range now so the start/end can be easily determined.